### PR TITLE
feat(options): add ancestry CSS selector to nodes

### PR DESCRIPTION
**Pull request by WilcoFiers**
Adds an option for axe.run() to return an "ancestry" css selector. Similar to "selector" but instead including just the node names and child index up to the root node of the tree. Handles iframe and shadow DOM nesting in the same way .selector does.

Closes issue: #2209

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [x] Follows the commit message policy, appropriate for next version
- [x] Code is reviewed for security

### CODE REVIEW

File: lib/core/utils/get-ancestry.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import getShadowSelector from './get-shadow-selector';
+
+function generateAncestry(node) {
```
Comment (straker) [inline lib/core/utils/get-ancestry.js:3]: I think we should cache these results on the virtual node as we process them so we aren't recalculating the entire ancestry tree of every descendant of a node.

File: lib/core/utils/get-ancestry.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	const parent = node.parentElement;
+	if (!parent) {
+		return nodeName;
+	}
+
+	let nthChild = '';
+	if (
+		nodeName !== 'head' &&
+		nodeName !== 'body' &&
+		parent.children.length > 1
```
Comment (straker) [inline lib/core/utils/get-ancestry.js:14]: Would it be better if this was only checking if the parent had multiple of the nodeName? I know `getSelector` only adds nth-child stuff if it needs to.

Right now if the parent has two children but they are both different nodeNames, you still get the nth-child selector which doesn't do much but add noise:

```
<html>
  <body>
    <div>
      <a href="/foo">Foo</a>
      <span>Hello</span>
    </div> 
  </body>
</html>
```

Results in `html > body > div:nth-child(1) > a:nth-child(1)`

File: test/core/reporters/helpers/process-aggregate.js [hunk 3]
```diff
@@ -69,6 +73,7 @@ describe('helpers.processAggregate', function() {
 							selector: '#dopel',
 							source: '<input id="dopel"/>',
 							xpath: '/main/input[@id="dopel"]',
+							ancestry: 'html > body > main > input:nth-child(0)',
```
Comment (straker) [inline test/core/reporters/helpers/process-aggregate.js:76]: how do we get an nth-child of 0?

File: lib/core/utils/get-ancestry.js (hunk 2 above)
Comment (WilcoFiers) [reply lib/core/utils/get-ancestry.js:14]: What would the benefit of that be? This seems faster.

**Pull request merged**